doc/api/url.md

```diff
@@ -2,139 +2,244 @@
 
     Stability: 2 - Stable
 
-This module has utilities for URL resolution and parsing.
-Call `require('url')` to use it.
+The `url` module provides utilities for URL resolution and parsing. It can be
+accessed using:
 
-## URL Parsing
+```js
+const url = require('url');
+```
 
-Parsed URL objects have some or all of the following fields, depending on
-whether or not they exist in the URL string. Any parts that are not in the URL
-string will not be in the parsed object. Examples are shown for the URL
+## URL Strings and URL Objects
 
-`'http://user:pass@host.com:8080/p/a/t/h?query=string#hash'`
+A URL String is a structured string containing multiple meaningful components. 
+When parsed, a URL Object is returned containing properties for each of these
+components.
 
-* `href`: The full URL that was originally parsed. Both the protocol and host are lowercased.
+The following details each of the components of a parsed URL. The example
+`'http://user:pass@host.com:8080/p/a/t/h?query=string#hash'` is used to
+illustrate each.
 
-    Example: `'http://user:pass@host.com:8080/p/a/t/h?query=string#hash'`
+```
++-----------------------------------------------------------------------------+
+| href                                                                        |
++----------+--+-----------+------------------+-------------------------+------+
+| protocol |* | auth      | host             | path                    | hash |
+|          |  |           +-----------+------+----------+--------------+      |
+|          |  |           | hostname  | port | pathname | search       |      |
+|          |  |           |           |      |          +-+------------+      |
+|          |  |           |           |      |          | | query      |      |
+" http:     //  user:pass @  host.com : 8080 / p/a/t/h  ? query=string #hash  "
+|          |  |           |           |      |          | |            |      |
++----------+--+-----------+-----------+------+----------+-+------------+------+
+(all spaces in the "" line should be ignored -- they're purely for formatting)
+*: given by "slashes" key
```

Comment:
This is confusing. It makes it sounds like the string literal `//` is what is stored under the `slashes` property, but in reality `slashes` is just a boolean indicating the existence of slashes after the protocol.